### PR TITLE
[buildkite] Add CI builds on GPUs

### PR DESCRIPTION
**Pull request by amontoison**
@sshin23

### CODE REVIEW

**Comment by sshin23** [comment]
test-self-hosted can be commented out

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      queue: "juliagpu"
+      cuda: "*"
+    env:
+      EXAMODELS_TEST_CUDA: "TRUE"
+    command: |
+      julia --color=yes -e '
+        using Pkg
+        Pkg.add("CUDA")
+        Pkg.add("KernelAbstractions")
+        Pkg.add("NLPModels")
```
Comment (sshin23) [inline .buildkite/pipeline.yml:16]: why was this necessary?

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -14,6 +14,13 @@ steps:
         Pkg.add("CUDA")
         Pkg.add("KernelAbstractions")
         Pkg.add("NLPModels")
+        Pkg.add("JuMP")
+        Pkg.add("ForwardDiff")
+        Pkg.add("NLPModelsIpopt")
+        Pkg.add("NLPModelsJuMP")
+        Pkg.add("Percival")
+        Pkg.add("PowerModels")
+        Pkg.add("MadNLP")
```
Comment (sshin23) [inline .buildkite/pipeline.yml:23]: not sure if this is ideal. Maybe we can create a separate Project.toml file for testing?

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (amontoison) [reply .buildkite/pipeline.yml:16]: It depends on how you want to run the tests.
If you just want to use `runtests.jl`, we can modify it.
But with the current `pipeline.yml`, we can also run specific and longer tests that target specific GPUs without environment variables.

I did something similar in `Krylov.jl`, using buildkite to run GPU tests separately from CPU tests, which is recommended to avoid occupying the GPUs for too long, allowing other users and repositories to access them.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (amontoison) [reply .buildkite/pipeline.yml:23]: The best solution is to create one `Project.toml` for each dependency. We can create a CI build for each extension if we do that.

Comment (sshin23) [reply .buildkite/pipeline.yml:23]: Can't we do, say, `Project.toml` on `/test` and have `Pkg.add("CUDA.jl")` and `Pkg.add("KernelAbstractions.jl")` before project environment instantiation?

Comment (amontoison) [reply .buildkite/pipeline.yml:23]: We can do that, it's what I do for KrylovPreconditioners.jl:
https://github.com/JuliaSmoothOptimizers/KrylovPreconditioners.jl/blob/main/.buildkite/pipeline.yml

Comment (sshin23) [reply .buildkite/pipeline.yml:23]: I think this approach is good. Can we do it in this way?

**Comment by sshin23** [comment]
Thanks @amontoison for setting this up!
